components/cdc/src/endpoint.rs

```diff
@@ -753,31 +815,162 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
                         }
                         return None;
                     }
-                        rx.await.unwrap_or(None)
+                    rx.await.unwrap_or(None)
+                }
+            })
+            .collect();
+        let resps = futures::future::join_all(regions).await;
+        resps
+            .into_iter()
+            .filter_map(|resp| resp)
+            .collect::<Vec<u64>>()
+    }
+
+    async fn region_resolved_ts_store(
```

Comment:
Could you move `region_resolved_ts_store` and `region_resolved_ts_raft` to a dedicated struct to decouple leadership confirming from Endpoint?

Reply:
> Should it be removed when leader step down?
Not necessary, it's expected to be updated after leader step down.

> Could you move `region_resolved_ts_store` and `region_resolved_ts_raft` to a dedicated struct to decouple leadership confirming from Endpoint?
It will be moved out from cdc crate in next PR.